builtin/providers/aws/resource_aws_emr.go

```diff
@@ -0,0 +1,439 @@
+	}
+	coreInstanceCount := d.Get("core_instance_count").(int)
+
+	applications := d.Get("applications").(*schema.Set).List()
+	var userKey, subnet, extraMasterSecGrp, extraSlaveSecGrp, emrMasterSecGrp, emrSlaveSecGrp, instanceProfile, serviceRole string
+	instanceProfile = "EMR_EC2_DefaultRole"
+	ec2Attributes := d.Get("ec2_attributes").([]interface{})
+	if len(ec2Attributes) == 1 {
+		attributes := ec2Attributes[0].(map[string]interface{})
+		userKey = attributes["key_name"].(string)
+		subnet = attributes["subnet_id"].(string)
+		extraMasterSecGrp = attributes["additional_master_security_groups"].(string)
+		extraSlaveSecGrp = attributes["additional_slave_security_groups"].(string)
+		emrMasterSecGrp = attributes["emr_managed_master_security_group"].(string)
+		emrSlaveSecGrp = attributes["emr_managed_slave_security_group"].(string)
+
+		if len(strings.TrimSpace(attributes["instance_profile"].(string))) != 0 {
+			instanceProfile = strings.TrimSpace(attributes["instance_profile"].(string))
+		}
+	}
+
+	if v, ok := d.GetOk("service_role"); ok {
+		serviceRole = v.(string)
+	} else {
+		serviceRole = "EMR_DefaultRole"
+	}
+
+	emrApps := expandApplications(applications)
+
+	params := &emr.RunJobFlowInput{
+		Instances: &emr.JobFlowInstancesConfig{
+			Ec2KeyName:                  aws.String(userKey),
+			Ec2SubnetId:                 aws.String(subnet),
+			InstanceCount:               aws.Int64(int64(coreInstanceCount + 1)),
+			KeepJobFlowAliveWhenNoSteps: aws.Bool(true),
+			MasterInstanceType:          aws.String(masterInstanceType),
+			SlaveInstanceType:           aws.String(coreInstanceType),
+			TerminationProtected:        aws.Bool(false),
+			AdditionalMasterSecurityGroups: []*string{
+				aws.String(extraMasterSecGrp),
+			},
+			AdditionalSlaveSecurityGroups: []*string{
+				aws.String(extraSlaveSecGrp),
+			},
+			EmrManagedMasterSecurityGroup: aws.String(emrMasterSecGrp),
+			EmrManagedSlaveSecurityGroup:  aws.String(emrSlaveSecGrp),
+		},
+		Name:         aws.String(d.Get("name").(string)),
+		Applications: emrApps,
+
+		JobFlowRole:       aws.String(instanceProfile),
+		ReleaseLabel:      aws.String(d.Get("release_label").(string)),
+		ServiceRole:       aws.String(serviceRole),
+		VisibleToAllUsers: aws.Bool(d.Get("visible_to_all_users").(bool)),
+	}
+
+	if v, ok := d.GetOk("log_uri"); ok {
+		logUrl := v.(string)
+		params.LogUri = aws.String(logUrl)
+	}
+	if v, ok := d.GetOk("bootstrap_action"); ok {
+		bootstrapActions := v.(*schema.Set).List()
+		log.Printf("[DEBUG] %v\n", bootstrapActions)
+		params.BootstrapActions = expandBootstrapActions(bootstrapActions)
+	}
+	if v, ok := d.GetOk("tags"); ok {
+		tagsIn := v.(map[string]interface{})
+		params.Tags = expandTags(tagsIn)
+	}
+	if v, ok := d.GetOk("configurations"); ok {
+		confUrl := v.(string)
+		params.Configurations = expandConfigures(confUrl)
+	}
+
+	log.Printf("[DEBUG] EMR Cluster create options: %s", params)
+	resp, err := conn.RunJobFlow(params)
```

Comment:
We need to poll the after creation process and validate that the EMR Cluster enters the state ([status defined here](http://docs.aws.amazon.com/ElasticMapReduce/latest/API/API_ClusterStatus.html)) we're expecting. I used [the example config from the test](https://github.com/hashicorp/terraform/pull/7619/files#diff-d53b8d8e18462f63430bdce6475994b2R82) and while Terraform reported everything was ok, I checked the console and after some time the console reported an error:

![emr_error](https://cloud.githubusercontent.com/assets/61721/17716468/69e1a446-63ce-11e6-9fa3-68d80f244a77.png)


See AWS DB for example:

- https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_db_instance.go#L603-L622